Applications/PaintBrush/main.cpp

```diff
@@ -96,7 +97,27 @@ int main(int argc, char** argv)
         return;
     }));
 
-    menubar->add_menu("Edit");
+    auto& edit_menu = menubar->add_menu("Edit");
+
+    auto paste_action = GUI::CommonActions::make_paste_action(
+        [&](const GUI::Action&) {
+            auto old_bitmap = RefPtr<Gfx::Bitmap>(paintable_widget.bitmap());
+            auto clipboard_bitmap = GUI::Clipboard::the().get_bitmap(old_bitmap->format(), old_bitmap->size());
```

Comment:
I don't think we should assume the old bitmap and the clipping bitmap have identical format and size. :)

Reply:
This is where it gets messy for me. I understand that the clipboard should hold info of format an size of the bitmap it has right now but I need a bit of time to think about how to do it properly.

Sorry about that. Noob things. Marking this as a draft to work on that!

Reply:
Okay so now clipboard knows the size and format of the last copied thing which makes applications stop assuming what size and format a bitmap from it is.

However I have to ask here because I couldn't find any solution for this. The contents of Clipboard::m_current_bitmap_format and Clipboard::m_current_bitmap_size are perfectly right if the same PaintBrush that copied into the clipboard tries to get the values.

However if some other PaintBrush (or application) tries to get_bitmap those 2 members always have default values. 

I'm assuming I'm copying values onto those members. Why are they dorking when other app calls the Clipboard singleton?

Sorry about the question ; I have the intuition that is just a trivial thing that I don't know.

Reply:
FML. They're freaking different applications. Ignore everything above.
![image](https://user-images.githubusercontent.com/6155245/80870273-ae4d1080-8ca5-11ea-8010-3b207eff97ef.png)

I'll try later to add a call to the WindowServer via IPC and save the bitmap properties on the WindowServer to then get them.

Reply:
Okay now it's ready. It stores the contents, format, width and height of the bitmap on the window server clipboard. not the client's one... duh!